custom_components/tesla_custom/select.py

```
@@ -370,6 +377,45 @@ def current_option(self):
+        if option == POWERWALL_MODE[0]:  # Charge from Grid
+            op_mode, grid_charging = "self_consumption", True
+        elif option == POWERWALL_MODE[1]:  # Discharge
+            op_mode, grid_charging = "self_consumption", False
+        elif option == POWERWALL_MODE[2]:  # Hold (use Grid)
+            op_mode, grid_charging = "autonomous", True
+        else:
+            return
+        try:
+            await self._energysite.set_operation_mode(op_mode)
+            await self._energysite.set_grid_charging(grid_charging)
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)